File: bitget-java-sdk-api/src/main/java/com/bitget/openapi/ws/BitgetWsClient.java
package com.bitget.openapi.ws;

import com.bitget.openapi.dto.request.ws.SubscribeReq;
import com.bitget.openapi.dto.request.ws.WsBaseReq;

import java.util.List;

public interface BitgetWsClient {

    void sendMessage(WsBaseReq<?> req);

    void sendMessage(String message);

    void unsubscribe(List<SubscribeReq> str);

    void subscribe(List<SubscribeReq> list);

    void subscribe(List<SubscribeReq> list, SubscriptionListener listener);

    void login();
}

File: bitget-php-sdk-api/src/internal/BitgetWsClient.php
<?php

namespace bitget\internal;

use bitget\model\ws\WsBaseReq;

interface BitgetWsClient
{
    public function sendMessage(WsBaseReq $baseReq): void;

    public function sendTextMessage(string $message): void;

    public function unsubscribe(array $req): void;

    public function subscribeDef(array $req): void;

    public function subscribe(array $req,Listener $listener): void;

}